### PR TITLE
Related items

### PR DESCRIPTION
**Pull request by callumreid**
-Optimized bundle.js size using uglification(Size before: 1.63MiB, size after: 352 KiB)
-Preloaded largest above the fold image in index.html (LCP before: 8s, after: 4.6s)
-Improved accessibility by adding lang='en' to HTML in index.html, and adding alt properties to images
-Created dark-mode toggle that creates a crude dark mode

Once merged, everyone will need to run `npm install uglifyjs-webpack-plugin --save-dev`
I added a script in package.json, buildProd. Running this instead of the normal build command runs webpack in production mode, which includes tree-shaking and uglification of JS.

### CODE REVIEW

File: client/dist/styles/darkMode.css [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.dark-app {
```
Comment (rlawrence9) [inline client/dist/styles/darkMode.css:1]: interesting this is how you do it 👀

File: client/dist/styles/relatedProducts.css [hunk 2]
```diff
@@ -62,6 +62,13 @@ REALTED PRODUCTS STYLING
   border: none;
 }
 
+.nextButton {
```
Comment (rlawrence9) [inline client/dist/styles/relatedProducts.css:65]: digging the rem 🔥

File: client/src/tests/relatedProductsTests.test.js [hunk 3]
```diff
@@ -46,7 +52,33 @@ describe('RelatedProducts methods', () => {
     const instance = wrapper.instance();
     expect(typeof instance.handleAddToOutfit).toBe('function');
   })
+})
 
+let rpcwrapper;
```
Comment (rlawrence9) [inline client/src/tests/relatedProductsTests.test.js:57]: if you were having trouble seeing the state of the instances after using mount .... Try either 

- mounting the component and creating the instance in beforeEach
- or wrapping the mount in a promise like this 
`      return new Promise((resolve, reject) => {
        component = renderer.create(
          <QnAClicks>
            {allClicksProps => {
              return <QuestionsNAnswers allClicksProps={allClicksProps} {...props}/>
            }}

          </QnAClicks>
        )
        resolve(component)
      })
      .then(newComponent => {
        let questions = newComponent.toJSON().children[0].children[4].children
        expect(questions[0].props.className).toEqual('showQuestion')
        expect(questions[1].props.className).toEqual('showQuestion')
        expect(questions[2].props.className).toEqual('hide')
      })`

File: package.json [hunk 4]
```diff
@@ -5,6 +5,7 @@
   "main": "index.js",
   "scripts": {
     "build": "webpack --mode development --watch",
+    "buildProd": "webpack --mode production --watch",
```
Comment (rlawrence9) [inline package.json:8]: ty 🤙

File: webpack.config.js [hunk 5]
```diff
@@ -1,18 +1,29 @@
 const path = require('path');
+const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
```
Comment (rlawrence9) [inline webpack.config.js:2]: ty 🤙

**Review by rlawrence9 — APPROVED** [review]

File: client/src/tests/relatedProductsTests.test.js (hunk 3 above)
Comment (callumreid) [reply client/src/tests/relatedProductsTests.test.js:57]: Thank you, I will look into this. Was having a hard time seeing anything, and never could get state

**Pull request merged**